fast_response/MonitoringAndMocks/Data_Display.py

```diff
@@ -8,756 +8,387 @@
 matplotlib.rcParams['ytick.labelsize'] = 16
 import matplotlib.pyplot as plt
 import numpy as np
-import glob, os, datetime, pwd, pickle
-from datetime import date
+import pickle
+import glob
+import os
+import datetime
+from datetime import date, timezone
+from dateutil.relativedelta import *
 from statistics import median
-#from statistics import mode
 from matplotlib.dates import DateFormatter
 from astropy.time import Time
 import pandas as pd
+from PIL import Image, ImageDraw, ImageFont
+# from IPython.display import HTML
+# from IPython.display import display
 import subprocess
 import warnings
 warnings.filterwarnings('ignore', module='astropy._erfa')
 
 def dial_up(who="jessie"):
-    cell_tower = "/home/jthwaites/private/"
-    subprocess.call([cell_tower+"make_call.py", f"--{who}=True", '--troubleshoot=True'])
+        cell_tower = "/home/jthwaites/private/"
+        subprocess.call([cell_tower+"make_call.py", f"--{who}=True", '--troubleshoot=True'])
 
-path=os.environ.get('FAST_RESPONSE_OUTPUT')
+path = os.environ.get('FAST_RESPONSE_OUTPUT')
 out_put = '/data/user/jthwaites/o4-mocks/'
 
 #Creating readable (and callable) files from ALL pickle files previously created in gw_gcn_listener
-if pwd.getpwuid(os.getuid())[0] == 'realtime':
-    Pickle_to_text = sorted(glob.glob(path+'PickledMocks/*MS*.pickle'))#+
-                            #glob.glob('/data/user/jthwaites/FastResponseAnalysis/output/PickledMocks/*MS*.pickle'))
-else:
-    Pickle_to_text = sorted(glob.glob(path+'PickledMocks/*MS*.pickle'))
-
-all_dictionary = {'Trigger_Time': [], 'GCN_Alert': [], 'End_Time': [],
-                  'LVK_Latency': [], 'IceCube_Latency': [], 'Total_Latency': [],
-                  'We_had_to_wait:': [], 'Name': [], 'Time_Stamp': []}
-print('Number of mocks to load: {}'.format(len(Pickle_to_text)))
-
-i=0
-for file in Pickle_to_text:
-    #if i%100==0: print(i)
-    if os.path.exists(file):
-        with open(file, 'rb') as pickle_now:
-            Entries = pickle.load(pickle_now)
-        if int(Entries["Trigger_Time"]) == 59957:
-            continue #This is January 13th 2023, dates were messed up.
-        for Key in Entries.keys():
-            if Key=="Ligo_Latency":
-                all_dictionary['LVK_Latency'].append(Entries[Key])
+mock_files = sorted(glob.glob(path+'PickledMocks/*MS*.pickle'))
```

Comment:
can we limit the number of mocks loaded here? This was more than 30,000 when I tried it. maybe just:
`mock_files = sorted(glob.glob(path+'PickledMocks/*MS*.pickle'), reverse=True)[:5000]`
to get the most recent 5000